src/RL/utils/geometricflux/heterogeneousgraphconv.jl

```diff
@@ -113,4 +113,93 @@ function (g::HeterogeneousGraphConv{<:AbstractMatrix,<:Any,meanPooling})(fg::Het
+        filteredembvaltovar = reshape(reduce(hcat ,filteredembvaltovar), :,size(H1)[2:3]...)
+        filteredembvartocon = reshape(reduce(hcat ,filteredembvartocon), :,size(H2)[2:3]...)
+        filteredembvartoval = reshape(reduce(hcat ,filteredembvartoval), :,size(H3)[2:3]...)
+
+    return BatchedHeterogeneousFeaturedGraph{Float32}(
+        contovar,
+        valtovar,
+        g.σ.(g.weightsvar ⊠ vcat(X1, H1, filteredembcontovar, filteredembvaltovar) .+ g.biasvar),
+        g.σ.(g.weightscon ⊠ vcat(X2, H2, filteredembvartocon) .+ g.biascon),
+        g.σ.(g.weightsval ⊠ vcat(X3, H3, filteredembvartoval) .+ g.biasval),
+        fgs.gf
+    )
+    end
+end
+"""
+function (g::HeterogeneousGraphConv{<:AbstractMatrix,<:Any, maxPooling})(fg::HeterogeneousFeaturedGraph, original_fg::HeterogeneousFeaturedGraph)
+
+This function operates the coordinate-wise max-Pooling technique along the neightbors of every node of the non-batched input HeterogeneousFeaturedGraph.
+
+For more details about the operations, please look at function (g::GraphConv{<:AbstractMatrix,<:Any, maxPooling})(fg::FeaturedGraph) in graphConv.jl. The same operations are done 4 times considering type-specific embeddings and adjacency matrix. 
+"""
+function (g::HeterogeneousGraphConv{<:AbstractMatrix,<:Any, maxPooling})(fg::HeterogeneousFeaturedGraph, original_fg::HeterogeneousFeaturedGraph)
+    contovar, valtovar = fg.contovar, fg.valtovar
+    vartocon, vartoval = transpose(contovar), transpose(valtovar)
+    
+    H1, H2, H3 = fg.varnf, fg.connf, fg.valnf
+    X1, X2, X3 = original_fg.varnf, original_fg.connf, original_fg.valnf
+    Zygote.ignore() do      
+        contovarIdx = repeat(collect(1:size(contovar,1)),1,size(contovar,2)).*contovar
+        valtovarIdx = repeat(collect(1:size(valtovar,1)),1,size(valtovar,2)).*valtovar
+        vartoconIdx = repeat(collect(1:size(vartocon,1)),1,size(vartocon,2)).*vartocon
+        vartovalIdx = repeat(collect(1:size(vartoval,1)),1,size(vartoval,2)).*vartoval
+
+        filteredcolcontovar = map(x-> filter(y -> y!=0,x),eachcol(contovarIdx))
+        filteredcolvaltovar = map(x-> filter(y -> y!=0,x),eachcol(valtovarIdx))
+        filteredcolvartocon = map(x-> filter(y -> y!=0,x),eachcol(vartoconIdx))
+        filteredcolvartoval = map(x-> filter(y -> y!=0,x),eachcol(vartovalIdx))
+
+        filteredembcontovar = mapreduce(x->!isempty(x) ? Base.maximum(H2[:,x], dims = 2) : zeros(Float32,size(H2,1)), hcat, filteredcolcontovar)
+        filteredembvaltovar = mapreduce(x->!isempty(x) ? Base.maximum(H3[:,x], dims = 2) : zeros(Float32,size(H3,1)), hcat, filteredcolvaltovar)
+        filteredembvartocon = mapreduce(x->!isempty(x) ? Base.maximum(H1[:,x], dims = 2) : zeros(Float32,size(H1,1)), hcat, filteredcolvartocon)
+        filteredembvartoval = mapreduce(x->!isempty(x) ? Base.maximum(H1[:,x], dims = 2) : zeros(Float32,size(H1,1)), hcat, filteredcolvartoval)
+    
+    return HeterogeneousFeaturedGraph(
```

Comment:
Operations here should be defined outside Zygote.ignore() to allow for backpropagation.